_inc/client/components/inline-expand/index.jsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import Gridicon from 'components/gridicon';
+
+export const InlineExpand = React.createClass( {
+
+	propTypes: {
+		label: React.PropTypes.string.isRequired,
+		icon: React.PropTypes.string,
+		cardKey: React.PropTypes.string,
+		expanded: React.PropTypes.bool,
+		onClick: React.PropTypes.func,
+		onClose: React.PropTypes.func,
+		onOpen: React.PropTypes.func
+	},
+
+	getInitialState: function() {
+		return {
+			expanded: this.props.expanded
+		};
+	},
+
+	getDefaultProps: function() {
+		return {
+			icon: '',
+			onOpen: () => false,
+			onClose: () => false,
+			cardKey: '',
+			expanded: false
+		};
+	},
+
+	onClick: function() {
+		if ( this.props.children ) {
+			this.setState( { expanded: ! this.state.expanded } );
+		}
+
+		if ( this.props.onClick ) {
+			this.props.onClick();
+		}
+
+		if ( this.state.expanded ) {
+			this.props.onClose( this.props.cardKey );
+		} else {
+			this.props.onOpen( this.props.cardKey );
+		}
+	},
+
+	render: function() {
+		let isExpanded = this.state.expanded
```

Comment:
Would be clearer to use the `classnames` package here. It's already a dependency in package.json

```
import classNames from 'classnames';
...
<div className={ classNames( 'jp-inline-expand', this.props.className, { 'is-expanded': this.state.expanded } ) }>
```